core/ArchiveProcessor/Loader.php

```diff
@@ -63,20 +64,32 @@ protected function mustProcessVisitCount($visits)
 
     public function prepareArchive($pluginName)
     {
-        $this->params->setRequestedPlugin($pluginName);
-
-        list($idArchive, $visits, $visitsConverted) = $this->loadExistingArchiveIdFromDb();
-        if (!empty($idArchive)) {
-            return $idArchive;
+        // temporarily set the idSite query parameter so archiving will end up using
+        // the correct site aware caches
+        $originalIdSite = isset($_GET['idSite']) ? $_GET['idSite'] : null;
```

Comment:
Looks better! I wonder what happens when someone posts `idSite` though... I think plugins that use `Common:getRequestVar('idSite')` eg in `addDimensions()` would still use the POST value. This might be an issue when archiving through webarchiving/browser archiving eg in an API request. Should be maybe also have some logic for this and check also POST?

Reply:
I think `$_GET + $_POST` will use `$_GET` over `$_POST`, and here we don't care if there's a `$_POST`, since we are really just passing the idSite to downstream code during archiving.

But it would definitely be safer to check for $_POST too, I'll add it.